file/file_prefetch_buffer.cc

```diff
@@ -268,16 +266,37 @@ void FilePrefetchBuffer::UpdateBuffersIfNeeded(uint64_t offset) {
     bufs_[second].buffer_.Clear();
   }
 
+  {
+    // In case buffers do not align, reset second buffer. This can happen in
+    // case readahead_size is set.
```

Comment:
This happened in case of db_stress because reads might not be sequential if readahead_size is set and we go for prefetching anyway. I tried adding logging but couldn't find the exact reason as to what exact condition is causing this issue. But the issue was buffers were not aligned. One had old data (related to prev_offset_) and other had new data (related to offset called).